File: BaseLibrary/command/miner/PutMineAnswerCommand.cs
using System;
using System.Collections.Generic;

namespace BaseLibrary.command.miner {
    public class PutMineAnswerCommand : AMinerCommand {

        private static readonly List<ISubCommandFactory> SUB_COMMAND_FACTORIES = new List<ISubCommandFactory>();

        public static int RegisterSubCommandFactory(ISubCommandFactory subCommandFactory) {
            int position = SUB_COMMAND_FACTORIES.Count;
            SUB_COMMAND_FACTORIES.Add(subCommandFactory);
            return position;
        }

        /// <summary>
        /// Mine id if SUCCESS is false.
        /// </summary>
        public const int FALSE_MINE_ID = 0;

        private bool _success;
        /// <summary>
        /// <list type="bullet">
        /// <item>
        ///     <description>true - mine was putted.</description>
        /// </item>
        /// <item>
        ///     <description>false - mine was not putted. Robot reach maximum of putted mines.</description>
        /// </item>
        /// </list>
        /// </summary>
        public bool SUCCESS {
            get {
                if (pending)
                    throw new NotSupportedException("Cannot access to property of pending request.");
                return _success;
            } private set =>_success = value; }

        private int _mineID;
        /// <summary>
        /// Mine id for putted mine.
        /// </summary>
        public int MINE_ID {
            get {
                if (pending)
                    throw new NotSupportedException("Cannot access to property of pending request.");
                return _mineID;
            } private set => _mineID = value; }

        /// <summary>
        /// Using for create pending command.
        /// </summary>
        public PutMineAnswerCommand() { }

        /// <summary>
        /// Using for create command fulled with data.
        /// </summary>
        public PutMineAnswerCommand(bool success, int mineId) {
            SUCCESS = success;
            MINE_ID = mineId;
            pending = false;
        }

        /// <summary>
        /// Fill command by data from another command (use full for filling pending command by command with data).
        /// </summary>
        /// <param name="source"></param>
        public void FillData(PutMineAnswerCommand source) {
            SUCCESS = source.SUCCESS;
            MINE_ID = source.MINE_ID;
            pending = false;
        }
    }
}

File: Java-Client/RobotBattlefield-Client-Java/src/BaseLibrary/communication/command/miner/PutMineAnswerCommand.java

package BaseLibrary.communication.command.miner;

import BaseLibrary.communication.protocol.AProtocol;

public class PutMineAnswerCommand extends AMinerCommand {

	private static final String NAME = "PUT_MINE_ANSWER";

	static {
		AProtocol.registerForDeserialize(NAME, PutMineAnswerCommand.class);
	}

	@Override
	public String getCommandName() {
		return NAME;
	}
	
	/**
	 * Mine id if SUCCESS is false.
	 */
	public static final int FALSE_MINE_ID = 0;

	private boolean success;
	private int mineID;

	public PutMineAnswerCommand() {
	}

	public PutMineAnswerCommand(boolean success, int mineID) {
		this.success = success;
		this.mineID = mineID;
		pending = false;
	}

	/**
	 * Fill command by data from another command (use full for filling pending
	 * command by command with data).
	 */
	public void FillData(PutMineAnswerCommand source) {
		success = source.success;
		mineID = source.mineID;
		pending = false;
	}

	/**
	 * <ul>
	 * <li>true - mine was put.</li>
	 * <li>false - mine was not put. Robot reach maximum of put mines.</li>
	 * </ul>
	 */
	public boolean isSuccess() {
		if (pending)
			throw new UnsupportedOperationException("Cannot access to property of pending request.");
		return success;
	}

	/**
	 * Mine id for put mine.
	 */
	public int getMineID() {
		if (pending)
			throw new UnsupportedOperationException("Cannot access to property of pending request.");
		return mineID;
	}

	public void setMineID(int mineID) {
		this.mineID = mineID;
		pending = false;
	}
}
